apidef/oas/xtyk_doc_test.go

```diff
@@ -0,0 +1,33 @@
+package oas
+
+import (
+	"flag"
+	"io/ioutil"
+	"testing"
+)
+
+var (
+	dumpXTykDocHelp = `if this flag is passed in tests, file "./schema/x-tyk-gateway.md"
+will be updated after TestExtractDocFromXTyk has passed`
+	dumpXTykDoc = flag.Bool("x-tyk-dump-doc", false, dumpXTykDocHelp)
+	xTykDocPath = "./schema/x-tyk-gateway.md"
+)
+
+func TestExtractDocFromXTyk(t *testing.T) {
+	fInfo, err := ExtractDocFromXTyk()
+	if err != nil {
+		if _, ok := err.(*DocErrList); ok {
```

Comment:
Should use `errors.As` - but, would we not just do an `assert.NoError(t, err)`? In case the error is nil, this test doesn't do an assertion, and each test should do a min(1) assertion.

Reply:
we chose to ignore this test for now since not every field is documented; and `errors.As` matches error values but here we are matching against a `type`

Reply:
@urbanishimwe small correction `errors.Is` matches values, `errors.As` matches type;

```
var docErr *DocumentError
if errors.As(err, &docErr) {
    // use docErr for assertions
} else {
    // ¯\_(ツ)_/¯
}
```

All good, since this is the `go generate` entrypoint too; Resolving it.